src/main/java/org/csanchez/jenkins/plugins/kubernetes/AllContainersRunningPodWatcher.java

```diff
@@ -89,6 +90,8 @@ boolean areAllContainersRunning(Pod pod) {
                     if (waitingStateMsg != null && waitingStateMsg.contains("Back-off pulling image")) {
                         LOGGER.log(Level.INFO, "Unable to pull Docker image");
 
+                        k8sLauncherTaskListener.error("Unable to pull Docker image. Check if image name is spelled correctly").flush();
```

Comment:
@jglick Is this what you meant by using `error()` via the TaskListener? We tried this and the message isn't being printed to the build console.

Reply:
Possibly it is being printed to the _agent_ log, rather than the _build_ log, and I suppose this plugin does not currently mirror agents logs to the build log even when the agent is being launched in response to a particular `Step` and is expected to be removed when that step completes. (Compare, for example, the much more direct `dockerNode` step in `docker-plugin`.) If so, then you need to somehow get a handle on the `StepExecution` (`PodTemplateStepExecution` I think), or at least its associated `TaskListener`.

As a reminder, this is a rather complex plugin which I have only occasionally contributed to. While I am intimately familiar with Pipeline, and pretty familiar with Kubernetes, there is a lot I do follow here in terms of the flow of control.